### DISCUSSION TITLE
How can I avoid yarn generating symlinks to my internal packages????

### DISCUSSION THREAD

**Discussion by MikeMitterer** [yarnpkg/berry#5801] [General]
This drives me crazy! In my .yarnrc.yml I've set `nmHoistingLimits: dependencies` witch works really nice and does what I want. The only exception is that yarn (4.0.0-rc.53) symlinks my internal packages. If one of these packages is not built on this machine (if there is no lib/dist folder) I get all kind of crazy errors.

<img width="1242" alt="Bildschirmfoto 2023-10-17 um 12 21 41" src="https://github.com/yarnpkg/berry/assets/116654/1b9e89ce-83d9-4024-b774-fac979afb2bf">

**Comment by clemyan** [comment]
I assume `packages/_ui/client/mobisim` is a workspace, and the symlinked packages (`@mmit/auth`, `@mmit/communication`, ...) are also workspaces in the same project?

If you want to fetch these packages externally (e.g. from npm) you can [disable transparent workspaces](https://yarnpkg.com/configuration/yarnrc#enableTransparentWorkspaces). Do note that this means changes to a local workspace won't be visible to other workspaces until you publish them externally (e.g. to npm). Any packages that you *do* want to symlink need to be declared using `workspace:`.

**Reply by MikeMitterer** [reply]
Thanks!
